### DISCUSSION TITLE
Custom Page Path in Nuxt 3

### DISCUSSION THREAD

**Discussion by svaldivia12** [nuxt/nuxt#17287] [Questions] [answered]
I am currently migrating a site to Nuxt 3. In Nuxt 2 I was using [router-extras](https://github.com/nuxt-community/router-extras-module#usage) module to define my custom routes.

Is there a way to override the path of a page in Nuxt 3?

**Comment by Flanker72** [comment]
This is how you can implement such logic: https://v3.nuxtjs.org/guide/directory-structure/pages#alias

Also you can define custom routing map using router options: https://v3.nuxtjs.org/guide/directory-structure/pages/#router-options

**Reply by svaldivia12** [reply]
Thanks for your response!

> This is how you can implement such logic: https://v3.nuxtjs.org/guide/directory-structure/pages#alias

I tried alias and worked correctly, but it adds a new path to the page, but it doesn't delete the old one. For example, if I have the page "TheLogin.vue", and I add the alias "/login", I can visit my login page using "/thelogin" and "/login". My idea is just to have the "/login" path.

> Also you can define custom routing map using router options: https://v3.nuxtjs.org/guide/directory-structure/pages/#router-options

Everything I tried with the router options failed. Can you provide some simple example or a link please?

**Reply by Flanker72** [reply]
Digging through source code I found that there is an ability to remove route: https://github.com/nuxt/framework/blob/8c2c80e43e80cd8f1c4246d7563214fa695a3e98/packages/nuxt/src/app/plugins/router.ts#L172
Maybe this can help clean your routing map :)

**Comment by svaldivia12** [comment, marked as answer]
A bit hacky but functional solution:

`nuxt.config.ts`
```
export default defineNuxtConfig({
   ...
   hooks: {
      'pages:extend': pages => {
         for (const page of pages) {
            if (page.name === 'TheLogin') {
               page.path = '/login'
               break
            }
         }
      },
   },
   ...
})
```

If someone have a better solution, please feel free to share it.
